pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -273,17 +281,81 @@ def _on_evaluation_epoch_end(self) -> None:
         self.trainer._call_lightning_module_hook("on_epoch_end")
         self.trainer.logger_connector.on_epoch_end()
 
-    def _print_results(self, results: List[_OUT_DICT], stage: RunningStage) -> None:
-        # TODO: this could be updated to look nicer
-        from pprint import pprint
-
-        print("-" * 80)
-        for i, metrics_dict in enumerate(results):
-            print(f"DATALOADER:{i} {stage.upper()} RESULTS")
-            pprint(
-                {
-                    k: (v.item() if v.numel() == 1 else v.tolist()) if isinstance(v, torch.Tensor) else v
-                    for k, v in metrics_dict.items()
-                }
-            )
-            print("-" * 80)
+    @staticmethod
+    def _get_keys(data: dict) -> Iterable[str]:
+        if any(isinstance(v, dict) for v in data.values()):
+            for v in data.values():
+                yield from apply_to_collection(v, dict, dict.keys)
+        else:
+            yield from data.keys()
+
+    @staticmethod
+    def _find_value(data: dict, target: str) -> Iterable[Any]:
+        for k, v in data.items():
+            if k == target:
+                yield v
+            elif isinstance(v, dict):
+                yield from EvaluationLoop._find_value(v, target)
+
+    @staticmethod
+    def _print_results(results: List[_OUT_DICT], stage: str, file: Optional[IO[str]] = None) -> None:
+        # remove the dl idx suffix
+        results = [{k.split("/dataloader_idx_")[0]: v for k, v in result.items()} for result in results]
+        metrics = sorted({k for keys in apply_to_collection(results, dict, EvaluationLoop._get_keys) for k in keys})
+        headers = [f"DataLoader {i}" for i in range(len(results))]
+
+        # fallback is useful for testing of printed output
+        term_size = shutil.get_terminal_size(fallback=(120, 30)).columns
+        max_length = int(min(max(len(max(metrics + headers, key=len)), 25), term_size / 2))
+
+        rows: List[List[Any]] = [[] for _ in metrics]
+
+        for result in results:
+            for metric, row in zip(metrics, rows):
+                v = list(EvaluationLoop._find_value(result, metric))
+                if v:
+                    val = v[0]
+                    if isinstance(val, torch.Tensor):
+                        val = val.item() if val.numel() == 1 else val.tolist()
+                    row.append(f"{val}")
+                else:
+                    row.append(" ")
+
+        # keep one column with max length for metrics
+        num_cols = int((term_size - max_length) / max_length)
```

Comment:
One job is failing:

```python
[1,0]<stderr>:  File "/Users/runner/work/pytorch-lightning/pytorch-lightning/pytorch_lightning/loops/dataloader/evaluation_loop.py", line 325, in _print_results
[1,0]<stderr>:    num_cols = int((term_size - max_length) / max_length)
[1,0]<stderr>:ZeroDivisionError: division by zero
```

Reply:
Where do you get this error? I believe it implies that terminal column size is zero:
```python
term_size = shutil.get_terminal_size(fallback=(120, 30)).columns
max_length = int(min(max(len(max(metrics + headers, key=len)), 25), term_size / 2))
```
We can solve this by adding that ```max_length``` should be at least equal to one, but not sure whether this makes sense with terminal column size being equal to zero?

Reply:
I'm not sure either. It's caused by Horovod tests: https://github.com/PyTorchLightning/pytorch-lightning/runs/5046435233?check_suite_focus=true#step:16:3236

It might be a problem with it and `shutil`. We can fall back to a 120 default in this case

Reply:
A quick search led me to:
- this issue https://bugs.python.org/issue42174 and
- this PR which includes a fix https://github.com/python/cpython/pull/29046

>    .. versionchanged:: 3.11
>       The ``fallback`` values are also used if :func:`os.get_terminal_size`
>       returns zeroes.

The fix/update will be shipped in 3.11, so we need to handle it in our side for the time being.

@AndresAlgaba Shall we use the default 120 if term_size==0 as @carmocca suggests?

Reply:
Thanks for looking into it @akihironitta! Seems good to me, added the ```or 120``` in the latest commit..